chain/xrp/client/client_test.go

```diff
@@ -547,42 +628,29 @@ func TestFetchTxInfo(t *testing.T) {
-				require.Equal(t, txInfo.Transfers[0].From[0].Balance.String(), "10000012")
-				require.Contains(t, txInfo.Transfers[0].To[0].Address, "chains/addresses/rLETt614usCXtkc8YcQmrzachrCaDjACjP")
-				require.Equal(t, txInfo.Transfers[0].To[0].Balance.String(), "10000000")
-
-				require.Equal(t, txInfo.Fees[0].Balance.String(), "12")
 
-			} else if testNo == 1 {
-				require.NoError(t, err)
-				require.NotNil(t, txInfo)
-
-				require.Equal(t, txInfo.Hash, "9D4D9CB01F4FFB12CA6262966311936B182E325A80461645E78EF54C11D2751B")
+			require.NoError(t, err)
+			require.NotNil(t, txInfo)
 
-				require.Contains(t, txInfo.Transfers[0].From[0].Address, "chains/addresses/rzvAXDKJnPi8m25HjXYiXAjJnzc7LGTfw")
-				require.Equal(t, txInfo.Transfers[0].From[0].Balance.String(), "4862478")
-				require.Contains(t, txInfo.Transfers[0].From[1].Address, "chains/addresses/rs9ineLqrCzeAGS1bxsrW8x2n3bRJYAh3Q")
-				require.Contains(t, txInfo.Transfers[0].From[1].Asset, "chains/assets/rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq")
-				require.Contains(t, txInfo.Transfers[0].From[1].Contract, "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq")
-				require.Equal(t, txInfo.Transfers[0].From[1].Balance.String(), "2624741712800000")
+			require.Equal(t, txInfo.Hash, vector.expectedTxInfo.Hash)
 
-				require.Contains(t, txInfo.Transfers[0].To[0].Address, "chains/addresses/rzvAXDKJnPi8m25HjXYiXAjJnzc7LGTfw")
-				require.Contains(t, txInfo.Transfers[0].To[0].Asset, "chains/assets/rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq")
-				require.Contains(t, txInfo.Transfers[0].To[0].Contract, "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq")
-				require.Equal(t, txInfo.Transfers[0].To[0].Balance.String(), "2624741712800000")
-				require.Contains(t, txInfo.Transfers[0].To[1].Address, "chains/addresses/rs9ineLqrCzeAGS1bxsrW8x2n3bRJYAh3Q")
-				require.Equal(t, txInfo.Transfers[0].To[1].Balance.String(), "4862466")
+			for idx, fromTransfer := range vector.expectedTxInfo.Transfers[0].From {
+				require.Equal(t, fromTransfer.Address, vector.expectedTxInfo.Transfers[0].From[idx].Address)
+				require.Equal(t, fromTransfer.Balance.String(), vector.expectedTxInfo.Transfers[0].From[idx].Balance.String())
+				if fromTransfer.Contract != "" {
+					require.Equal(t, fromTransfer.Contract, vector.expectedTxInfo.Transfers[0].From[idx].Contract)
+				}
+			}
 
-				require.Equal(t, txInfo.Fees[0].Balance.String(), "12")
+			for idx, toTransfer := range vector.expectedTxInfo.Transfers[0].To {
+				require.Equal(t, toTransfer.Address, vector.expectedTxInfo.Transfers[0].To[idx].Address)
+				require.Equal(t, toTransfer.Balance.String(), vector.expectedTxInfo.Transfers[0].To[idx].Balance.String())
+				if toTransfer.Contract != "" {
+					require.Equal(t, toTransfer.Contract, vector.expectedTxInfo.Transfers[0].To[idx].Contract)
+				}
 			}
 
+			require.Equal(t, txInfo.Fees[0].Balance.String(), vector.expectedTxInfo.Fees[0].Balance.String())
```

Comment:
why only comparing a subset of the first transfer? 

I would expect to do:

```go
require.Equal(t, vector.expectedTxInfo, txInfo)
```

This is simpler and also tests all of the fields (e.g. block, confirmations, etc).

Alternatively you could break it down a bit:

```go
   // compare block info 
   require.Equal(t, vector.expectedTxInfo.Block, txInfo.Block)
   
   // compare other fields
  // ...
   
   // compare all tfs
  for i := range txInfo.Transfers {
    require.Equal(t, vector.expectedTxInfo.Transfers[i], txInfo.Transfers[i])
  }
```

but in both cases we test everything.